payloads/v9/channel.ts

```diff
@@ -79,37 +78,45 @@ export interface APIChannel extends APIPartialChannel {
 	 * The absence of this field in API calls and Gateway events should indicate that slowmode has been reset to the default value.
 	 */
 	rate_limit_per_user?: number;
+}
+
+export interface APITextChannel extends APITextBasedChannel {
+	type: ChannelType.GuildText;
 	/**
-	 * The recipients of the DM
-	 *
-	 * See https://discord.com/developers/docs/resources/user#user-object
+	 * Default duration for newly created threads, in minutes, to automatically archive the thread after recent activity
 	 */
-	recipients?: APIUser[];
+	default_auto_archive_duration?: ThreadAutoArchiveDuration;
 	/**
-	 * Icon hash
+	 * The channel topic (0-1024 characters)
 	 */
-	icon?: string | null;
+	topic?: string | null;
+}
+
+export interface APINonRateLimitedChannel extends Omit<APITextChannel, 'rate_limit_per_user' | 'type'> {
```

Comment:
This should be called APINewsChannel

Making the type a generic means you don't have to omit it here